src/main/java/com/aws/greengrass/status/model/Trigger.java

```diff
@@ -11,8 +11,7 @@ public enum Trigger {
     LOCAL_DEPLOYMENT,
     THING_DEPLOYMENT,
     THING_GROUP_DEPLOYMENT,
-    ERRORED_COMPONENT,
-    BROKEN_COMPONENT,
+    COMPONENT_STATUS_CHANGE,
```

Comment:
Add the new field in the middle of the enum will cause issue with older nucleus versions, which are not aware of this field. This new enum item should be added to the end of the list.

Enums use ordering internally to generate `int` as ordinals for comparision. 
When enums are marshalled, the int is used to the digest. Same goes for any comparision. 
In this case; ordinal value of `COMPONENT_STATUS_CHANGE` is 5, but for older nucleus without this trigger type, ordinal 5 is `RECONNECT`.

Example playground code: https://onecompiler.com/java/425mnfkbw

Java spec: https://docs.oracle.com/javase/8/docs/api/java/lang/Enum.html#ordinal--

Particular line of interest: _"The natural order implemented by this method is the order in which the constants are declared."_

Reply:
That is only true if you're using the ordinal somwhere. We should always be passing enums as strings

Reply:
Ack. 

`.toString` will return the name of the enum in this case. 

Because we use Jackson to marshal the data before sending it out of the process, we should be good.